test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/RequestHeadersTimeoutTests.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests
+{
+    public class RequestHeadersTimeoutTests
+    {
+        private static readonly TimeSpan RequestHeadersTimeout = TimeSpan.FromSeconds(10);
+        private static readonly TimeSpan LongDelay = TimeSpan.FromSeconds(30);
+        private static readonly TimeSpan ShortDelay = TimeSpan.FromSeconds(LongDelay.TotalSeconds / 10);
+
+
+        [Fact]
+        public async Task TestRequestHeadersTimeout()
+        {
+            using (var server = CreateServer())
+            {
+                var tasks = new[]
```

Comment:
The only issue I have with this suggestion is that they would run sequentially in OS X, since we've disabled parallelism there. These are long running tests and they can all run against the same server/app, that's why I wrote these tests this way (same with keep-alive tests).